packages/ifpack2/src/Ifpack2_RILUK_def.hpp

```diff
@@ -926,9 +926,10 @@ void RILUK<MatrixType>::compute ()
         A_local_crs_nc->fillComplete (A_local_->getDomainMap (), A_local_->getRangeMap ());
         A_local_crs = rcp_const_cast<const crs_matrix_type> (A_local_crs_nc);
       }
-      A_local_rowmap_  = A_local_crs->getLocalMatrixDevice().graph.row_map;
-      A_local_entries_ = A_local_crs->getLocalMatrixDevice().graph.entries;
-      A_local_values_  = A_local_crs->getLocalValuesView();
+      auto lclMtx = A_local_crs->getLocalMatrixDevice();
+      A_local_rowmap_  = lclMtx.graph.row_map;
+      A_local_entries_ = lclMtx.graph.entries;
+      A_local_values_  = lclMtx.values;
```

Comment:
@kddevin Thank you for these changes.  For my own education, was this change made in order to reduce data transfer, or was the old approach deprecated and/or result in errors?

Reply:
@jhux2   CrsMatrix used to store a pointer to a pre-constructed KokkosSparse::CrsMatrix, and getLocalMatrix() returned that pointer.  After the UVM changes, CrsMatrix no longer stores a KokkosSparse::CrsMatrix but, rather, builds one on demand when getLocalMatrixDevice/Host is called.  So this change reduces the number of times the KokkosSparse::CrsMatrix is constructed.  The old method would still work, however.  Another alternative would be
```
A_local_rowmap_ = A_local_crs->getLocalRowPtrsDevice();
A_local_entries = A_local_crs->getLocalIndicesDevice();
A_local_values_ = A_local_crs->getLocalValuesDevice(Tpetra::Access::ReadWrite);
```
Note that if A_local_values_ is actually used as read-only in this Ifpack2 case, using the latter pattern with Tpetra::Access::ReadOnly might reduce data transfer, as the modified flag would not be set.

Are you seeing any problems?